app/src/main/java/net/squanchy/tweets/util/TwitterDateUtils.java

```diff
@@ -0,0 +1,105 @@
+package net.squanchy.tweets.util;
+
+import android.content.Context;
+import android.content.res.Resources;
+import android.text.format.DateUtils;
+
+import com.twitter.sdk.android.core.models.Tweet;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.Locale;
+
+import net.squanchy.R;
+import net.squanchy.support.lang.Optional;
+
+public class TwitterDateUtils {
+
+    // Sat Mar 14 02:34:20 +0000 2009
+    private static final SimpleDateFormat DATE_TIME_RFC822 = new SimpleDateFormat("EEE MMM dd HH:mm:ss Z yyyy", Locale.ENGLISH);
+    private static final SimpleDateFormat RELATIVE_DATE_FORMAT = new SimpleDateFormat("MM/dd/yy", Locale.ENGLISH);
+    private static final long INVALID_DATE = -1;
+
+    private TwitterDateUtils() {
+    }
+
+    public static String getTimestampFrom(Tweet tweet, Context context) {
+        return getOptionalTimestamp(tweet.createdAt, context)
+                .or(context.getString(R.string.tweet_date_not_available));
+    }
+
+    private static Optional<String> getOptionalTimestamp(String date, Context context) {
+        final String formattedTimestamp;
+
+        if (date != null && TwitterDateUtils.isValidTimestamp(date)) {
+            final Long createdAtTimestamp = TwitterDateUtils.apiTimeToLong(date);
+            formattedTimestamp = TwitterDateUtils.getRelativeTimeString(context.getResources(),
+                    System.currentTimeMillis(), createdAtTimestamp);
+        } else {
+            formattedTimestamp = null;
+        }
+
+        return Optional.fromNullable(formattedTimestamp);
+    }
+
+    private static long apiTimeToLong(String apiTime) {
+        if (apiTime == null) {
+            return INVALID_DATE;
+        }
+
+        try {
+            return DATE_TIME_RFC822.parse(apiTime).getTime();
+        } catch (ParseException e) {
+            return INVALID_DATE;
+        }
+    }
+
+    private static boolean isValidTimestamp(String timestamp) {
+        return apiTimeToLong(timestamp) != INVALID_DATE;
+    }
+
+    /**
+     * This method is not thread safe. It has been modified from the original to not rely on global
+     * time state. If a timestamp is in the future we return it as an absolute date string. Within
+     * the same second we return 0s
+     *
+     * @param res               resource
+     * @param currentTimeMillis timestamp for offset
+     * @param timestamp         timestamp
+     * @return the relative time string
+     */
+    private static String getRelativeTimeString(Resources res, long currentTimeMillis, long timestamp) {
```

Comment:
The warning here makes me shiver ❌ 

Since we don't even need to have a relative time span (examples are all "[today] at [11:42]" so let's just assume we can do away with a simple format:
 1. `today` or `yesterday` or a date (such as `3rd May 2017`) if older than that
 2. ` on `
 3. the time, such as `11:42`

Besides, this seems to be basically doing what `DateUtils` do already. Use `net.danlew.android.joda.DateUtils` if possible, instead of copying this from somewhere else (Twitter SDK?).